SalesforceCommerceCloud/pwa-kit
# Generator fix pwa-kit.config template path

# Description



Fixes the path to obtain the `pwa-kit.config` template in the `pwa-kit-create-app/scripts/create-mobify-app.js` script. The template includes the configuration to not include the locale in the URL as part of generating a project with a single-locale.

The previous solution assumed that the `create-mobify-app.js` script was always run from the root of the monorepo.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Use require instead of path.join to read the pwa-kit.config template.

# How to Test-Drive This PR

1. Checkout this branch `generator-single-locale-fix-path`.
2. Run `npm ci`.
3. Generate a project locally from the root of the monorepo by running the `create-mobify-app` script:
```
GENERATOR_PRESET=test-project node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/project1
```
4. Generate a project locally from `packages/pwa-kit-create-app` by running the `create-mobify-app` script:
```
GENERATOR_PRESET=test-project node ./scripts/create-mobify-app-dev.js --outputDir /tmp/project2
```
5. Confirm the two resulted generated projects have this URL configuration for the locale type in the `pwa-kit.config.js` file:
```
{
  "url": {
    "locale": "none"
  }
}
```

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
